### PR TITLE
cmake: fix failure due missing thrift build scripts if building with jaeger

### PR DESCRIPTION
**Pull request by ideepika**
this shall address the missing failure,

CMake Error at cmake/modules/BuildJaeger.cmake:61 (include):
  include could not find load file:

    Buildthrift

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ideepika** [comment]
this shall fix the failures seen in: https://shaman.ceph.com/builds/ceph/wip-debian-jaeger-kefu/85e731d9e0cf115eac57305ae13684339fbc7de2/

**Comment by ideepika** [comment]
testing at: https://shaman.ceph.com/builds/ceph/wip-jaeger-buildthrift-deepika/84b53613cb41bf4a9462717435d92c26ff2c2d8c/

**Comment by ideepika** [comment]
https://shaman.ceph.com/builds/ceph/wip-jaeger-buildthrift-deepika/97a36368201b3cfd5285826251a062a5b3867131/

File: cmake/modules/Findthrift.cmake [hunk 1]
```diff
@@ -17,14 +17,11 @@ find_library(
...
-  set_target_properties(
-    thrift::libthrift
-    PROPERTIES IMPORTED_LOCATION ${thrift_LIBRARIES}
-               INTERFACE_INCLUDE_DIRECTORIES ${thrift_INCLUDE_DIR})
-endif()
+add_library(thrift::libthrift UNKNOWN IMPORTED)
+set_target_properties(
+  thrift::libthrift
+  PROPERTIES IMPORTED_LOCATION ${thrift_LIBRARIES}
+	     INTERFACE_INCLUDE_DIRECTORIES ${thrift_INCLUDE_DIR})
```
Comment (tchaikov) [inline cmake/modules/Findthrift.cmake:24]: you might want to transpose the `if(thrift_FOUND..)` block with the `find_package_handle_standard_args(thrift...)` statement, which sets `thrift_FOUND`.

**Review by tchaikov — DISMISSED** [review]

File: cmake/modules/Findthrift.cmake (hunk 1 above)
Comment (ideepika) [reply cmake/modules/Findthrift.cmake]: @tchaikov thanks, locally seems to do the trick, will test

File: cmake/modules/BuildJaeger.cmake [hunk 2]
```diff
@@ -58,8 +58,6 @@ function(build_jaeger)
   endif()
   include(BuildOpenTracing)
   build_opentracing()
-  include(Buildthrift)
```
Comment (tchaikov) [inline cmake/modules/BuildJaeger.cmake:61]: this is not a "cleanup". i think you are addressing the regression introduced by 80e82686ebafe36fca6dfd21cb32e63ced94d5cd. could you summarize the change more accurately?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -427,7 +427,7 @@ add_library(common-objs OBJECT ${libcommon_files})
 add_dependencies(common-objs legacy-option-headers)
 
 if(WITH_JAEGER)
-  find_package(thrift 0.13.0)
+  find_package(thrift 0.13.0 REQUIRED)
```
Comment (tchaikov) [inline src/CMakeLists.txt:430]: > make thrift 0.13 find_pkg REQUIRED if jaeger enabled

i find it is difficult to parse this line. a better title might be

> cmake: always require thrift v0.13.0 if WITH_JAEGER

File: cmake/modules/Findthrift.cmake [hunk 4]
```diff
@@ -17,16 +17,15 @@ find_library(
   HINTS ${thrift_HOME} ENV thrift_HOME /usr/local /opt/local
   PATH_SUFFIXES lib lib64)
 
+include(FindPackageHandleStandardArgs)
```
Comment (tchaikov) [inline cmake/modules/Findthrift.cmake:20]: this change should belong to a separated commit.

**Comment by tchaikov** [comment]
i'd like to see at least a single teuthology job completed with this change.

**Comment by ideepika** [comment]
@tchaikov tests finally passing: https://shaman.ceph.com/builds/ceph/wip-jaeger-buildthrift/9579aa9e958b644b109444b168b551ccc020b4ec/

**Comment by ideepika** [comment]
> i'd like to see at least a single teuthology job completed with this change.

ack, will do

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-07-24_05:22:39-rados-wip-kefu-testing-2021-07-24-1158-distro-basic-smithi/

**Review by tchaikov — APPROVED** [review]

**Comment by ideepika** [comment]
> i'd like to see at least a single teuthology job completed with this change.

@tchaikov thanks for testing and merging, sorry I misunderstood it to be testing with jaeger enabled earlier

**Comment by tchaikov** [comment]
@ideepika i think the bottom line is not to break the regular tests. but, IIUC, your goal is to enable Ceph to use jaeger, so probably it's still worthwhile to test with jaeger enabled.

**Comment by ideepika** [comment]
> @ideepika i think the bottom line is not to break the regular tests. but, IIUC, your goal is to enable Ceph to use jaeger, so probably it's still worthwhile to test with jaeger enabled.

@tchaikov ack, yep that's correct, working on it but it might need a little longer :) working on supporting --flavor args correctly for teuthology, so that we can pass crimson/jaeger(build flavors) by command line if not specified in the script: https://github.com/ceph/teuthology/compare/master...ideepika:wip-flavor?expand=1
